bundles/org.openhab.binding.myq/README.md

```diff
@@ -0,0 +1,71 @@
+# MyQ Binding
+
+This binding integrates with the [The Chamberlain Group MyQ](https://www.myq.com) cloud service. It allows monitoring and control over [MyQ](https://www.myq.com) enabled garage doors manufactured by LiftMaster, Chamberlain and Craftsman.
+
+## Supported Things
+
+### Account
+
+This represents the MyQ cloud account and uses the same credentials needed when using the MyQ mobile application.
+
+ThingTypeUID:`account`
+
+### Garage Door
+
+This represents a garage door associated with an account.  Multiple garage doors are supported.
+
+ThingTypeUID:`garagedoor`
+
+### Lamp
+
+This represents a lamp associated with an account.  Multiple lamps are supported.
+
+ThingTypeUID:`lamp`
+
+## Discovery
+
+Once an account has been added, garage doors and lamps will automatically be discovered and added to the inbox.
+
+## Channels
+
+| Channel       | Item Type     | Thing Type       | Description              |
+|---------------|---------------|------------------|--------------------------|
+| status        | String        | garagedoor, lamp | Status of the device     |
+| switch        | Switch        | garagedoor, lamp | ON (open), OFF (close)   |
+| contact       | Contact       | garagedoor       | Open, Closed             |
+| rollershutter | Rollershutter | garagedoor       | UP (open), DOWN (closed) |
```

Comment:
Is there any reason why these are two separate channels? What's the difference to the `switch` channel?

Reply:
Some users prefer this to be a switch, similar to how locks are represented, others consider the door more like a rollershutter.  I tried to support both use cases.

Reply:
There should be only 1 channel. Contact conceptually is readonly. Rollesshutter seems the most logically,but should accept percentage and move/stop. So that should be handled. See also https://www.openhab.org/docs/configuration/items.html#type

Reply:
thanks @Hilbrand for the feedback
>Rollesshutter seems the most logically,but should accept percentage and move/stop.

I would certainly like to , but since the api only responds to open/close and not a % or stop, and does not report %, how would you suggest i do that? 

>There should be only 1 channel.

Ideally that would be awesome, but its not very black and white what item type is in use here. From a UI perspective, a rollershutter makes the most sense,  our widgets specifically have a UP/Down interface, but as I mentioned above, we can not support % or stop.  From a contact point of view this also makes sense, and in fact is useful when using this in a group of contacts with sensors  for perimeter monitoring (which i do) , but of course there is not a command type for a contact, so that can not be the only channel.  Many users have used a simple switch to control this.  I am trying to appease the current users of previous myq bindings that have been floating around.

Is there another way to map multiple item types to a single channel?  In this case, i think the types i have here all make very logical sense, but picking only one will greatly limit the usability to our existing myq users. 

i am also taking a queue from other bindings,  like our Open Garage addon, which handles this in a similar way.   

I am open to suggestions, but i'm hoping this is not going to delay getting this 1.x replacement merged in.

Reply:
Regarding the rollershutter, you are correct i was not handling % commands and should have, so i have added that in my last commit.

Reply:
I go for the Contact type too, but what is the use case for the Switch in addition to the Rollershutter?

Reply:
I can live without the switch, it seemed to me a common UI element that users of the previous bindings were using, but a rollershutter covers this use case nicely.  I personally don't use it, so I'll remove it and see if anyone screams.

Reply:
I use the switch, please don't remove it. Other bindings have dimmers and on/offs for the same thing. I don't see how less choice is good thing.

Reply:
I'm ok leaving it in , although dimmers and switches are a different comparison, dimmer items also support ON/OFF commands and states, so you can map a Switch Item  to a dimmer channel and it will work, so i don't know a lot of bindings that do both channel types for the same function, or at least they should not.

Reply:
I also use the switch.  It just makes some things cleaner across my rules.

Reply:
I left the switch there in my last commit, i vote we keep it for consistency with existing user's expectations.

Reply:
Just putting in my two cents - I’ve used a myQ plugin of some form for a few years now. When I first came to it, I was extremely confused that their were so many channels that seemingly did the same thing. Especially since the Rollershutter channel didn’t respond to 0/100 which is what some things (HomeKit) automatically do instead of UP/DOWN. I saw you fixed that last part recently, and that’s nice. There’s just something to be said for presenting the simplest possible interface (with full functionality), and making users adapt it via rules if they want to use it say in a group with slightly different item types. As long as documentation is clear that these channels all actually do the same thing, I won’t argue against keeping them though.

Reply:
IMHO the current number of Channels for the same function are too much and therefore confusing. As several users seem to use the Switch, my suggestion would be to keep this and remove all others. As the garage door cannot be controlled with 0-100%, but only open/close, the Rollershutter doesn't have any benefit over the Switch. Keep it simple... If the user wants to do some more complex things, it's still possible by using rules. I think this fits good with openHAB concepts.

Reply:
I would agree with getting rid of the rollershutter and the contact.  Personally I use the switch and the status channels.  The switch to send commands, and the status for my rules (so that I can identify open vs. opening vs. closing. vs. closed).  If you really want it to say up/down, you can easily do that in the sitemap.

Reply:
So i feel very strongly about keeping a roller shutter.  I never once refer to my door as being ON or OFF.   In our UIs the rollershutter widget makes sense, an on/off switch does not and is confusing IMHO (why is a door that is open ON? ).    

>cannot be controlled with 0-100%

The binding does support 0%/100% as well as UP/DOWN, i don't think is uncommon for simple  roller shutters to  work this way, i have a unit which only goes up or down, not a %, i believe the motor simply inverts direction when the current is inverted. 

 < If the user wants to do some more complex things, it's still possible by using rules. 

If we are already working around an issue like this with rules before we have even submitted the binding i think we have lost from a usability POV.  

I would like  drop the contact channel and keep the others as it is, i do feel strongly about this.  My reasoning is there a small amount of channels in the binding,  i don't think it's going to confuse users, there are other bindings that do this, and  the fact that we have so many opinions on this i think proves the value of keeping both channels types .

Reply:
I'm cool with that.